docs/Tutorial_DiscordMERLIN.ipynb

```diff
@@ -0,0 +1,1965 @@
+{
```

Comment:
@seanlaw 
Thanks for the effort. I am working on this one....I will let you know how it goes...

Reply:
@seanlaw 
It works...I will provide some notes in the notebook so that I do not forget (they are not important for now)

**Just two things:**

* I think we do not have to linearly increase decay. (otherwise, we need to keep track of `number-of-attempts`. It is not hard...but it just add a couple of lines to the code).  So, we can simply do what you suggested... `min_dist = min_dist *  decay[len(discords_idx)]`, where `decay`'s values can be `0.99`, for instance.

*And...for padding...I first decided to pad with np.nan or -1... but I finally did this:
```
def _discords(...):
    # some code
    
    out = np.concatenate((discords_idx, discords_dist, discords_nn_idx), dtype=object).reshape(3,len(discords_idx)).transpose()
    
    n_pad = k - len(discords_idx)
    out = np.pad(out, ((0,n_pad),(0,0)), mode='constant', constant_values=-1)
    out[len(discords_idx):k,1] = np.NINF
    
    return out
```

#as you mentioned, I think np.NINF might be better than -1 for distance. Please let me know if you have any feedback.

Reply:
<p>No, I was thinking that we check <code>len(discords_idx)</code> like:</p><p> </p><pre class="ql-syntax" spellcheck="false">for i in range(max(k - len(discords, 0))):
    discords_idx.append(-1)
    discords_dist.append(np.NINF)
    discords_nn_idx.append(-1)

out = np.concatenate((discords_idx, discords_dist, discords_nn_idx), dtype=object).reshape(3,len(discords_idx)).transpose()
</pre>

Reply:
I see. Since you used `max`, I assume you wanted to avoid the case where `k<len(discords_idx)`....which means you are considering a case where `len(discords_idx)` becomes more than `k` (either due to a bug or due a design or ...)

So...should I return `out[:k]` at the end then?

Reply:
<p>Yes, precisely but, realistically, we should better figure out how to ensure that the length of discords_idx (and others are correct) in the code above all of this.</p><p> </p><p>Btw, to be a bit more efficient and avoid for-loops, maybe it is better to do:</p><p> </p><pre class="ql-syntax" spellcheck="false">repeat = max(k - len(discords, 0))
discords_idx.append([-1] * repeat)
discords_dist.append([np.NINF] * repeat)
discords_nn_idx.append([-1] * repeat)
</pre>